### DISCUSSION TITLE
Handlebars and boolean value

### DISCUSSION THREAD

**Discussion by tw02d008** [Budibase/budibase#10055] [Help] [answered]
I am frustrated with handlebars and bool values.
Is there a way in Handlebars to Update a state to True or False that will be recognizes as boolean?

What I try to achieve is the following:
I have a State called "historyMode" and I want to update via conditional UI settings my UI.
To toggle the state I have a button with the following handlebar code:

![image](https://user-images.githubusercontent.com/37417853/225618553-7d3a663d-67ad-4336-9e08-a72ddebd1874.png)

![image](https://user-images.githubusercontent.com/37417853/225619775-4f112997-bd58-431e-914d-2ce2e1c40135.png)

true and false is not recognized by the conditional UI:

![image](https://user-images.githubusercontent.com/37417853/225618913-0b107022-5abc-42e1-9e8b-96bc19ed0fbe.png)

Is there a way to set somehow boolean values in Handlebars or do I have to use Javascript?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @tw02d008,

I did some debugging and recreating and the below is the result of my findings.

To directly answer your question no I don't think handle bars should be used. The issue here I believe is with handle bar's handling of boolean values. It returns them as a string meaning isTruthy will always return false after it has been changed from false to true or vice versa.

This isn't something that I would use handlebars for as it always returns strings, making it very difficult to achieve your desired effect with handle bars. I would always opt for using Javascript whenever working with somewhat complicated conditions, objects, numbers and boolean values.

I'd recommend using the convert to Javascript button. It won't convert everything properly, so some debugging and fixing may be required.

I hope this helps with answering your question!

**Reply by melohagan** [reply]
@tw02d008  If you prefer using Handlebars, then you could also change the conditions to be: *Equals* **Text** False
